lib/fog/sakuracloud/models/volume/disk.rb

```diff
@@ -20,8 +20,12 @@ def delete
         alias_method :destroy, :delete
 
         def save
-          requires :name, :plan, :source_archive
-          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
+          requires :name, :plan
+          options = {
+            :source_archive=>@attributes[:source_archive],
+            :size_mb=>@attributes[:size_mb]
+          } 
+          data = service.create_disk(@attributes[:name], @attributes[:plan], options).body["Disk"]
```

Comment:
I will change instance variables to use the getters.
Regarding return of method `create_disk`, a testcase `./tests/sakuracloud/requests/volume/disks_tests.rb`, requires `disks.body.is_a?`
See following codes:

./tests/sakuracloud/requests/volume/disks_tests.rb
```
Shindo.tests('Fog::Volume[:sakuracloud] | create_disks request', ['sakuracloud', 'volume']) do
  tests('success') do
    tests('#create_disks') do
      disks = sakuracloud_volume_service.create_disk('foobar', 4, {:sourcearchive => 112500463685})
      test 'returns a Hash' do
        disks.body.is_a? Hash
      end
```